JumpscaleLibs/sal/zosv2/Zosv2.py

```diff
@@ -0,0 +1,197 @@
+
+    def check_ip_in_network(self, ip , network):
+        return netaddr.IPAddress(ip) in netaddr.IPNetwork(network)
+
+    def key_pair_get(self):
+        wg_private = public.PrivateKey.generate()
+        wg_public = wg_private.public_key
+
+        wg_private_base64 = wg_private.encode(Base64Encoder)
+        wg_public_base64 = wg_public.encode(Base64Encoder)
+
+        return wg_private_base64, wg_public_base64
+
+    def create_network(self,reservation,ip_range,node_id):
+        network = reservation.data_reservation.networks.new()
+        network.node_id = node_id
+        network.workload_id = 1
+        network.name = j.data.idgenerator.generateXCharID(16)
+        network.iprange = ip_range
+        return reservation , network
+
+    def access_node(self, network, ip_range, node):
+        network_range = netaddr.IPNetwork(ip_range).ip
+        # add 256 to plus 1 in network from 172.20.0.0 -> 172.20.1.0
+        network_range += 256
+        network1 = str(network_range) + "/24"
+        _, wg_private_encrypted, wg_public = j.tools.wireguard.generate_zos_keys(node.public_key_hex)
+        wg_port = self.find_free_wg_port(node)
+
+        nr1 = network.network_resources.new()
+
+        nr1.iprange = network1
+        nr1.node_id = node.node_id
+        nr1.wireguard_listen_port = wg_port
+        nr1.wireguard_public_key = wg_public
+        nr1.wireguard_private_key_encrypted = wg_private_encrypted
+
+        # add 256 to plus 1 in network from 172.20.0.0 -> 172.20.1.0
+        network_range += 256
+        network2 = str(network_range) + "/24"
+        ip = netaddr.IPAddress(network_range)
+        allowed_ip = ip.words
+
+        private_key, public_key = self.key_pair_get()
+        # add a peer to the network, this peer is your laptop
+        peer = nr1.peers.new()
+        peer.iprange = network2
+
+        peer.allowed_iprange = ["100.64." + str(allowed_ip[1]) + "." + str(allowed_ip[2]) + "/32", network2]
+        # this is your wireguard public key from your laptop
+        peer.public_key = public_key.decode()
+
+        self.config.public_key = public_key
+        self.config.private_key = private_key
+        self.config.wg_port = wg_port
+
+        return network, self.config
+
+    def add_volume(self,reservation, node_id, voulme_size=5, volume_type="SSD"):
+        volume = reservation.data_reservation.volumes.new()
+        volume.workload_id = 2
+        volume.size = voulme_size
+        volume.type = volume_type
+        volume.node_id = node_id
+        return  volume
+
+    def get_all_ips(self, ip_range):
+        networks = netaddr.IPNetwork(ip_range)
+        ips = []
+        for ip in list(networks.iter_hosts()):
+           ips.append(ip.format())
+        return ips
+
+    def create_container(self,network_name,ip_address,node_id, reservation, flist, env={}, entrypoint="",interactive=False, volume_size=0, voulme_type="SSD", mount_point="/sandbox/var"):
```

Comment:
All the argument regarding volumes are not correct. What if the user needs to add 2 volumes to its container ?
a better way would be to have a method that link a volume to a container, so the user has full fexibility.
It would do something like:
```python
vol1 = j.sal.zosv2.add_volume(....)
vol2 = j.sal.zosv2.add_volume(....)
cont = j.sal.zosv2.create_container(...)
j.sal.sozv2.attach_volume(cont, vol1, mountpoint)
j.sal.sozv2.attach_volume(cont, vol2, mountpoint)
```